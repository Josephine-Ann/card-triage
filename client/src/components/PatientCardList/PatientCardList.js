import './PatientCardList.css'
import React, { useContext } from 'react'
import { PatientInfoContext } from '../../PatientInfoProvider'
import { PatientCard } from '../PatientCard';


export const PatientCardList = () => {
    const {
        state: {
            patientList,
            doneIs,
            pendingIs,
            rejectedIs
        }
    } = useContext(PatientInfoContext);
    return (
        <div id="list__three-columns">
            <div id="list__columns">
                <div className={pendingIs ? 'list' : 'list-hidden'}>
                    <div className="list__titles-desktop">
                        <p className="list__titles-top">Pending Tests</p>
                    </div>
                    {
                        patientList.map((patient) => {
                            if (patient.status === 'PENDING') {
                                return <PatientCard key={patient.id} {...patient} />
                            }
                        })
                    }
                </div>
                <div className={doneIs ? 'list' : 'list-hidden'}>
                    <div className="list__titles-desktop">
                        <p className="list__titles-top">Tests Done</p>
                    </div>
                    {
                        patientList.map((patient) => {
                            if (patient.status === 'DONE') {
                                return <PatientCard key={patient.id} {...patient} />
                            }
                        })
                    }
                </div>
                <div className={rejectedIs ? 'list' : 'list-hidden'}>
                    <div className="list__titles-desktop">
                        <p className="list__titles-top">Tests Rejected</p>
                    </div>
                    {
                        patientList.map((patient) => {
                            if (patient.status === 'REJECTED') {
                                return <PatientCard key={patient.id} {...patient} />
                            }
                        })
                    }
                </div>
            </div>
        </div >
    )
}


export default PatientCardList