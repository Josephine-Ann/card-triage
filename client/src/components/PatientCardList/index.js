export { PatientCardList } from './PatientCardList.js'
