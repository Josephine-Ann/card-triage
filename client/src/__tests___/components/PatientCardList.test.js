import React from 'react'
import { shallow } from 'enzyme';
import { PatientCardList } from '../../components/PatientCardList/PatientCardList';
import patients from '../fixtures/patients';

jest.mock("react", () => ({
    ...jest.requireActual("react"),
    useContext: () => ({
        state: {
            patientList: [{
                "arrhythmias": [
                    "AFib",
                    "AV Block",
                    "Pause",
                    "PSVC",
                    "PVC"
                ],
                "created_date": "2020-03-10T13:14:59+0000",
                "id": 0,
                "patient_name": "Anita",
                "status": "PENDING"
            }]
        },
        // Mock the function to do nothing
        updateToDo: jest.fn(),
    }),
}));



describe("PatientCardList", () => {
    it("renders as expected in snapshot", () => {
        const wrapper = shallow(<PatientCardList />);
        expect(wrapper).toMatchSnapshot();
    });
});