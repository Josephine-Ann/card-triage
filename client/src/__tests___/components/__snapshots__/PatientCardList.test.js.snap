// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PatientCardList renders as expected in snapshot 1`] = `
ShallowWrapper {
  "length": 1,
}
`;
